### DISCUSSION TITLE
How can I do equivalent of torch.chunk in ggml tensor ?

### DISCUSSION THREAD

**Discussion by mnlife** [ggml-org/ggml#778] [Q&A]
I want using chatglm3-6b's model in llama.cpp, how can I do equivalent of torch.chunk in ggml tensor ?
﻿
```
# Project to 4h. If using swiglu double the output width, see https://arxiv.org/pdf/2002.05202.pdf
def swiglu(x):
    x = torch.chunk(x, 2, dim=-1)
    return F.silu(x[0]) * x[1]
```
this code from https://hf-mirror.com/THUDM/chatglm3-6b/blob/main/modeling_chatglm.py

**Comment by balisujohn** [comment]
You can accomplish this using `ggml_view_1d`
````
ggml_tensor * conditioning_scale = ggml_view_1d(ctx0, model.diffusion_conditioning_latent, 1024, 0); 
ggml_tensor * conditioning_shift = ggml_view_1d(ctx0, model.diffusion_conditioning_latent, 1024, ggml_element_size(model.diffusion_conditioning_latent) * 1024);

````
For example splitting a tensor with 2048 elements in half.

The first int argument is the number of elements, the second argument is the byte offset.
